content/en/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -504,6 +504,33 @@ stops adjusting the target (and sets the `ScalingActive` Condition on itself
 to `false`) until you reactivate it by manually adjusting the target's desired
 replica count or HPA's minimum replica count.
 
+### Migrating Deployments and StatefulSets to horizontal autoscaling
+
+When an HPA is enabled, it is recommended that the value of `spec.replicas` of
+the Deployment and/or Statefulset object is removed.  Without doing so, anytime
+a change to that object is applied, example via `kubectl apply -f
+deployment.yaml`, will instruct Kubernetes to scale the current number of Pods
+to the desired value retained in the `spec.replicas` key. This may not be
+desired and could be troublesome when an HPA is active.
+
+Keep in mind that the removal of `spec.replicas` may incur a one time
+degradation of Pod counts as the default value of this key is 1 (reference
+[Deployment Replicas](/docs/concepts/workloads/controllers/deployment#replicas).
+Upon the update all Pods except 1 will begin their Termination procedures.  Any
+deployment application afterwards will behave as normal and respect a Rolling
+Update configuration as desired.  We can avoid this degradation using two
+options:
+
+- Manually edit the last applied API configuration following the following
+   instructions:
+  1. `kubectl apply edit-last-applied deployment/<deployment_name>`
+  2. In the editor, remove `spec.replicas`
+  3. In your configuration file, you can now remove `spec.replicas`
+  4. From here on out you can run `kubectl apply -f deployment.yaml`
+- Utilize the flag `--server-side` when running `kubectl apply` - This will
```

Comment:
If they are using server-side apply, then they'll see a conflict once the HPA kicks in for the first time. Once they see a conflict, they can remove the replicas field without a problem. This behavior is nice because you'll never hit that scale-to-1 behavior: either there's no conflict yet and your old value takes, or there is a conflict, your apply is prevented, and it's safe to remove the field from your config. However, it's possible to avoid your main deploy pipeline hitting that conflict in the first place. There's already a guide for this, as it turns out: https://kubernetes.io/docs/reference/using-api/server-side-apply/#transferring-ownership

Reply:
:+1: for hyperlinking to https://kubernetes.io/docs/reference/using-api/server-side-apply/#transferring-ownership